src/BackupExplorer/Parser/BackupParserImpl.cs

```diff
@@ -33,9 +33,36 @@ public BackupParserImpl(string backupChainPath, string codePackagePath)
             this.backupChainPath = backupChainPath;
             this.codePackage = new CodePackageInfo(codePackagePath);
             this.workFolder = Path.Combine(Directory.GetCurrentDirectory(), Guid.NewGuid().ToString());
+            this.checkpointThreshold = 200;
 
             Console.WriteLine("Work Folder : {0}", this.workFolder);
+            InitializeBackupParser();
+           
+        }
+
+        public BackupParserImpl(string backupChainPath, string codePackagePath, ILog log, string workFolderPath, int checkpointThreshold = 200)
```

Comment:
can there be just 1 constructor with default/null values for other params ?

Reply:
Simplified to a single constructor with defaults